primitives/src/merkle_tree/namespaced_merkle_tree.rs

```diff
@@ -181,10 +188,297 @@ where
     }
 }
 
+type InnerTree<E, H, T, N, Arity> =
+    MerkleTree<E, NamespacedHasher<H, E, u64, T, N>, u64, Arity, NamespacedHash<T, N>>;
+
+#[derive(Debug, Clone)]
```

Comment:
Yes but we'll probably implement a custom serialization scheme so that we don't send the whole tree over the wire, tracked here: https://github.com/orgs/EspressoSystems/projects/20/views/4?pane=issue&itemId=28910685